lighthouse-core/config/budget.js

```diff
@@ -132,6 +132,25 @@ class Budget {
     return path;
   }
 
+  /**
+   * Returns the budget that applies to a given URL.
+   * If multiple budgets match based on thier 'path' property,
+   * then the last-listed of those budgets is returned.
+   * @param {Array<LH.Budget>|null} budgets
+   * @param {string} url
+   * @return {LH.Budget | undefined} budget
+   */
+  static matchingBudget(budgets, url) {
+    // Clones budget so that the user-supplied version is not mutated.
+    /** @type {Array<LH.Budget>} */
+    const clonedBudgets = JSON.parse(JSON.stringify(budgets || []));
+
+    // Applies the LAST matching budget
+    return clonedBudgets ? clonedBudgets.reverse().find((b) => {
```

Comment:
didn't notice this last time but we can get rid of the ternary if we're `|| []`ing in the clone step :)

Reply:
maybe a test for `null` budget would be nice too actually now that its living on the budget parsing class

Reply:
what about just, like
```js
static matchingBudget(budgets, url) {
  if (budgets === null) return;

  // Applies the LAST matching budget.
  for (let i = budgets.length - 1; i >= 0; i--) {
    const budget = budgets[i];
    if (Budget.urlMatchesPattern(url, budget.path)) {
      return budget;
    }
  }
}
```